contracts/LiquidationManager.sol

```diff
@@ -0,0 +1,63 @@
+pragma solidity ^0.8.25;
+
+import { ExponentialNoError } from "./Utils/ExponentialNoError.sol";
+
+/**
+ * @title LiquidationManager
+ * @dev This contract provides functions to manage liquidations in the venus protocol.
+ * It calculates close factors, dynamic liquidation incentives, and the number of tokens to seize during liquidation.
+ * It also checks if a liquidation is toxic based on average liquidation threshold and health factor.
+ * @author Venus
+ * @notice This contract is designed to be used in conjunction with the venus protocol's liquidation process.
+ */
+contract LiquidationManager is ExponentialNoError {
+    /**
+     * @notice Calculate the close factor for a liquidation
+     * @param borrowBalance The borrow balance of the borrower
+     * @param wtAvg The weighted average of the collateral
+     * @param totalCollateral The total collateral available for liquidation
+     * @param dynamicLiquidationIncentive The dynamic liquidation incentive, scaled by 1e18
+     * @param maxLiquidationIncentive The maximum liquidation incentive allowed
+     * @return closeFactor The calculated close factor, scaled by 1e18
+     */
+    function calculateCloseFactor(
+        uint256 borrowBalance,
+        uint256 wtAvg,
+        uint256 totalCollateral,
+        uint256 dynamicLiquidationIncentive,
+        uint256 maxLiquidationIncentive
+    ) external pure returns (uint256 closeFactor) {
+        if (dynamicLiquidationIncentive == maxLiquidationIncentive) {
+            // Prevent underflow
+            require(
+                wtAvg * totalCollateral <= borrowBalance * mantissaOne,
+                "LiquidationManager: Collateral exceeds borrow capacity"
+            );
+
+            uint256 numerator = borrowBalance * mantissaOne - wtAvg * totalCollateral;
+            uint256 denominator = borrowBalance * (mantissaOne - ((wtAvg * maxLiquidationIncentive) / mantissaOne));
+
+            closeFactor = numerator / denominator;
+            closeFactor = closeFactor > mantissaOne ? mantissaOne : closeFactor;
+        } else {
+            closeFactor = mantissaOne; // Liquidate 100% if unhealthy
+        }
+    }
+
+    /**
+     * @notice Calculate the dynamic liquidation incentive based on health factor and average liquidation threshold
+     * @param healthFactor The health factor of the borrower
+     * @param liquidationThresholdAvg The average liquidation threshold of the collateral
+     * @param maxLiquidationIncentiveMantissa The maximum liquidation incentive allowed, scaled by 1e18
+     * @return incentive The calculated dynamic liquidation incentive, scaled by 1e18
+     */
+    function calculateDynamicLiquidationIncentive(
+        uint256 healthFactor,
+        uint256 liquidationThresholdAvg,
+        uint256 maxLiquidationIncentiveMantissa
+    ) external pure returns (uint256 incentive) {
+        uint256 value = ((healthFactor * mantissaOne) / liquidationThresholdAvg) - mantissaOne;
```

Comment:
I think this value is always lower than 1, but `maxLiquidationIncentiveMantissa` is always greater than 1. I think we shouldn't substract `mantissaOne` here

Reply:
[5c84b79](https://github.com/VenusProtocol/venus-protocol/pull/604/commits/5c84b796d7eca4a79ca6835a30049480c38e4119)